### PR TITLE
Avoiding unnecessary get calls

### PR DESCRIPTION
**Pull request by amitbhanja**

### CODE REVIEW

File: code/src/nuvla/api/api.py [hunk 1]
```diff
@@ -360,7 +360,6 @@ def cloud_entry_point(self):
         if self._cimi_cloud_entry_point is None:
             self._cimi_cloud_entry_point = self._cimi_get_cloud_entry_point()
         return self._cimi_cloud_entry_point
-
```
Comment (schaubl) [inline code/src/nuvla/api/api.py:363]: Missing new line

Comment (amitbhanja) [reply code/src/nuvla/api/api.py:363]: Done

File: code/src/nuvla/api/api.py [hunk 2]
```diff
@@ -494,7 +492,7 @@ def edit(self, resource_id, data, **kwargs) -> CimiResource:
         :rtype:     CimiResource
         """
         resource = self.get(resource_id=resource_id)
-        operation_href = self._cimi_find_operation_href(resource, 'edit')
+        operation_href = resource.id
         return CimiResource(self._cimi_put(resource_id=operation_href, json=data, params=kwargs))
```
Comment (konstan) [inline code/src/nuvla/api/api.py:496]: IMO, having `operation_href` var is unnecessary. I suggest directly providing `resource.id` to the `resource_id` kwarg.

File: code/src/nuvla/api/api.py [hunk 3]
```diff
@@ -508,7 +506,7 @@ def delete(self, resource_id) -> CimiResponse:
 
         """
         resource = self.get(resource_id=resource_id)
-        operation_href = self._cimi_find_operation_href(resource, 'delete')
+        operation_href = resource.id
         return CimiResponse(self._cimi_delete(resource_id=operation_href))
```
Comment (konstan) [inline code/src/nuvla/api/api.py:510]: IMO, having `operation_href` var is unnecessary. I suggest directly providing `resource.id` to the `resource_id` kwarg.

File: code/src/nuvla/api/api.py [hunk 4]
```diff
@@ -567,7 +565,7 @@ def add(self, resource_type, data):
         :rtype:     CimiResponse
         """
         collection = self.search(resource_type=resource_type, last=0)
-        operation_href = self._cimi_find_operation_href(collection, 'add')
+        operation_href = f'{collection.id}'
         return CimiResponse(self._cimi_post(resource_id=operation_href, json=data))
```
Comment (konstan) [inline code/src/nuvla/api/api.py:570]: Why f-string needs to be used? Is `id` on the `collection` not of type str? And also, I don't see a need for the intermediary `operation_href` var.

File: code/src/nuvla/api/api.py [hunk 5]
```diff
@@ -379,9 +379,7 @@ def _cimi_get_uri(self, resource_id=None, resource_type=None):
             raise TypeError("You can only specify 'resource_id' or 'resource_type', not both.")
 
         if resource_type is not None:
-            resource_id = self.cloud_entry_point.collections.get(resource_type)
-            if resource_id is None:
-                raise KeyError("Resource type '{0}' not found.".format(resource_type))
+            resource_id = resource_type
 
         return resource_id
```
Comment (konstan) [inline code/src/nuvla/api/api.py:385]: Instead, the above can be done as `return resource_id or resource_type`.

**Review by konstan — CHANGES_REQUESTED** [review]

**Pull request merged**